siliconcompiler/tools/vpr/_json_constraint.py

```diff
@@ -0,0 +1,98 @@
+import argparse
+import json
+
+from siliconcompiler.tools.vpr._xml_constraint import generate_vpr_constraints_xml
+from siliconcompiler.tools.vpr._xml_constraint import write_vpr_constraints_xml_file
+
+
+def main():
+
+    option_parser = argparse.ArgumentParser()
+    option_parser.add_argument("-constraints_map",
+                               help="architecture-specific mapping of constraint pins "
+                                    "to FPGA core pins")
+    option_parser.add_argument("json_constraints",
+                               help="architecture-independent constraints file")
+    option_parser.add_argument("constraints_file_out",
+                               help="constraints XML file name")
+
+    options = option_parser.parse_args()
+
+    json_constraints_file = options.json_constraints
+    constraints_map_file = options.constraints_map
+    constraints_file_out = options.constraints_file_out
+
+    json_generic_constraints = load_json_constraints(json_constraints_file)
+    if (constraints_map_file):
+        constraints_map = load_constraints_map(constraints_map_file)
+    else:
+        constraints_map = {}
+
+    mapped_constraints = map_constraints(json_generic_constraints,
+                                         constraints_map)
+
+    constraints_xml = generate_vpr_constraints_xml(mapped_constraints)
+    write_vpr_constraints_xml_file(constraints_xml, constraints_file_out)
```

Comment:
probably not needed since this file is not meant to be run outside of the pre_processing step

Reply:
I used it for testing purposes during development; would only recommend keeping if we feel that such a test mode is useful in some future enterprise.

Reply:
just move this code down to the `if __name__...` section and remove the `main()` then